hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/DeletingServiceMetrics.java

```diff
@@ -0,0 +1,336 @@
+  private MutableGaugeLong numDirsSentForPurge;
+  @Metric("Total no. of sub-directories sent for purge")
+  private MutableGaugeLong numSubDirsSentForPurge;
+  @Metric("Total no. of sub-files sent for purge")
+  private MutableGaugeLong numSubFilesSentForPurge;
+
+  public void incrNumDirDeleted(long dirDel) {
+    numDirsSentForPurge.incr(dirDel);
+  }
+
+  public void incrNumDirsMoved(long dirMove) {
+    numSubDirsSentForPurge.incr(dirMove);
+  }
+
+  public void incrNumFilesMoved(long filesMove) {
+    numSubFilesSentForPurge.incr(filesMove);
+  }
+
+  public void incrementDirectoryDeletionTotalMetrics(long dirDel, long dirMove, long filesMove) {
+    incrNumDirDeleted(dirDel);
+    incrNumDirsMoved(dirMove);
+    incrNumFilesMoved(filesMove);
+  }
+
+  /*
+   * Directory deletion metrics in the latest iteration of DirectoryDeletingService.
+   */
+  @Metric("Iteration run count of DirectoryDeletingService")
+  private MutableGaugeLong iterationDirRunCount;
+  @Metric("Iteration start time of DirectoryDeletingService")
+  private MutableGaugeLong iterationDirStartTime;
+  @Metric("Total time taken by the last iteration of DirectoryDeletingService")
+  private MutableGaugeLong iterationDirDuration;
+  @Metric("No. of directories deleted in last iteration")
+  private MutableGaugeLong iterationDirsDeleted;
+  @Metric("No. of sub-directories deleted in last iteration")
+  private MutableGaugeLong iterationSubDirsDeleted;
+  @Metric("No. of sub-directories sent for purge in last iteration")
+  private MutableGaugeLong iterationSubDirsSentForPurge;
+  @Metric("No. of files sent for purge in last iteration")
+  private MutableGaugeLong iterationFilesSentForPurge;
+
+  public void setIterationDirRunCount(long runcount) {
+    iterationDirRunCount.set(runcount);
+  }
+
+  public void setIterationDirStartTime(long startTime) {
+    iterationDirStartTime.set(startTime);
+  }
+
+  public void setIterationDirDuration(long duration) {
+    iterationDirDuration.set(duration);
+  }
+
+  public void setIterationDirDeleted(long dirDel) {
+    iterationDirsDeleted.set(dirDel);
+  }
+
+  public void setIterationSubDirDeleted(long subdirDel) {
+    iterationSubDirsDeleted.set(subdirDel);
+  }
+
+  public void setIterationFilesSentForPurge(long filesMove) {
+    iterationFilesSentForPurge.set(filesMove);
+  }
+
+  public void setIterationSubDirsSentForPurge(long subdirMove) {
+    iterationSubDirsSentForPurge.set(subdirMove);
+  }
+
+  public void setDirectoryDeletionIterationMetrics(long runcount, long startTime, long duration,
+                                                   long dirDel, long subdirDel,
+                                                   long filesMove, long subdirMove) {
+    setIterationDirRunCount(runcount);
+    setIterationDirStartTime(startTime);
+    setIterationDirDuration(duration);
+    setIterationDirDeleted(dirDel);
+    setIterationSubDirDeleted(subdirDel);
+    setIterationFilesSentForPurge(filesMove);
```

Comment:
iteration level merics not required as its logged and overall metrics is there and information can be derived from that.